core/blockchain.go

```diff
@@ -1800,6 +1816,9 @@ func (bc *BlockChain) insertChain(chain types.Blocks, setHead bool) (int, error)
 			parent = bc.GetHeader(block.ParentHash(), block.NumberU64()-1)
 		}
 		statedb, err := state.New(parent.Root, bc.stateCache, bc.snaps)
+		if bc.collectWitness {
+			statedb.EnableWitnessCollection()
+		}
```

Comment:
Instead of having this as a separate mathod, pls pass it as a parameter to statedb.StartPrefetcher on line 1831

Reply:
Yes, that would mean witness building doesn't run for pre-byzantium, but we shouldn't really cater future features for irrelevant past chain segments.